### ISSUE TITLE
One to One links should only be written if there is not an existing entry with the same values

### ISSUE THREAD

**Issue by jegelstaff** [jegelstaff/formulize#364]
If the developer has manually created the matching one-to-one entry already for some reason, the one to one code that writes entries automatically, should not write a duplicate entry. But we don't check whether there's an entry already in the DB, we just check if we wrote one through the normal writing process (by checking the GLOBALS list of written entries). If someone used the API to write the entry directly, then the one to one process would write a duplicate. So we should check for an entry that has the matching values before we write it, to avoid creating duplicates.

**Comment by jegelstaff** [comment]
Lower priority

**Comment by jegelstaff** [comment]
I'm pretty sure there is checking???

**Comment by jegelstaff** [comment]
There is sort of checking. We only write to entries if the entry does not already have a value for the element. But we will write new entries if there appears to be no existing entry in the pair for that form. And we do only check this based on the entries written this page load through the normal process. We don't look in the database to see if there is any matching entry already for a given entry that we're working with, when only one form is active/present (which might happen if the screen has a one-to-one connection in its relationship, but only one form represented on this page).

So basically, this needs a complex setup with only one form represented on the page, and then experiment to see if an entry can be written and not have the match saved in the database?
